File: src/analysis.ts
/**
 * analysis — the mechanical half of the assessment license.
 *
 * Owns: nothing. Pure candidate GENERATORS, never verdicts: each verb
 * computes, deterministically, the population that may be judged. Whoever
 * judges receives candidates and never goes hunting — the feed is the
 * boundary that keeps "likely" out of the record.
 *
 * THE LICENSE ATTACHES TO THE ACTIVITY, NOT AN AGENT (ruled 2026-08-26,
 * artifact comment): the librarian may judge a feed itself — an analytical
 * question is just a question with a license attached — and dispatching a
 * separate analyst is a COST decision like any other delegation. Either
 * way the license rules hold: propose-only, grounds must resolve,
 * candidatesReceived === candidatesAssessed in the attestation, a feed's
 * gap is reported as a generator gap, never filled by invention.
 */
export const VERBS = ["pain-synthesis", "control-coverage", "conflict-support", "handoff-friction"] as const;
export type AnalysisVerb = (typeof VERBS)[number];

export interface Candidate { verb: AnalysisVerb; where: string; material: string; sources: string[]; }

/** the candidate feed for one verb, deterministic, engagement-scoped */
export function feeds(root: string, verb: AnalysisVerb): Candidate[] { throw new Error("mock-out"); }
/** THE artifact identity rule for edge matching, in one place */
export function normalizeArtifact(item: string): string { throw new Error("mock-out"); }

File: consult/analysis.py
"""analysis — the mechanical half of the analyst license.

Owns: nothing. Pure candidate GENERATORS, never verdicts (the oracle's
M39/M49 license, kept whole): each feed computes, deterministically, the
population the analyst is allowed to judge. The analyst receives candidates;
it never goes hunting. candidates_received must equal candidates_assessed in
its attestation — the feed is the boundary that keeps "likely" out of the
record.

Reframed per charter D6: these feeds fire when the human asks an analytical
question, not at a pipeline milestone. The librarian picks the verb(s) a
question implies, runs the feeds, dispatches the analyst over them, and
brings the proposals back to the conversation.

The four verbs (each one feed, each with its boundary):
  pain-synthesis      every pain callout verbatim with its sources
  control-coverage    steps that produce output and declare no control
  conflict-support    every gap that is a conflict on its face
  handoff-friction    orphan outputs and shared inputs across IPO edges
"""

from __future__ import annotations
from pathlib import Path

VERBS = ("pain-synthesis", "control-coverage", "conflict-support", "handoff-friction")


def feeds(root: Path, verb: str) -> list[dict]:
    """The candidate feed for one verb, deterministic, engagement-scoped."""
    raise NotImplementedError


def normalize_artifact(item: str) -> str:
    """THE artifact identity rule for edge matching, in one place (oracle-ported)."""
    raise NotImplementedError
